Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -57,7 +57,10 @@ public class PythonCompiler : SinglePassCompiler {
 
     public override void EmitCallToMain(Method mainMethod, string baseName, ConcreteSyntaxTree wr) {
       Coverage.EmitSetup(wr);
-      wr.WriteLine("_module._default.Main()");
+      var tryBlock = wr.NewBlockPy("try:");
+      tryBlock.WriteLine("_module._default.Main()");
+      var exceptBlock = wr.NewBlockPy("except _dafny.HaltException as e:");
+      exceptBlock.WriteLine($"_dafny.print(\"[Program halted] \" + str(e) + \"\\n\")");
```

Comment:
```suggestion
      wr.NewBlockPy("except _dafny.HaltException as e:")
        .WriteLine("_dafny.print(\"[Program halted] \" + str(e) + \"\\n\")");
```

Reply:
Yeah, I like this version better than the first one, where different output lines are different source lines too.